build-logic/build-update-utils/src/main/kotlin/gradlebuild/buildutils/tasks/AbstractCheckOrUpdateContributorsInReleaseNotes.kt

```diff
@@ -71,18 +67,31 @@ abstract class AbstractCheckOrUpdateContributorsInReleaseNotes : DefaultTask() {
     @Internal
     protected
     fun getContributorsInReleaseNotes(): Set<GitHubUser> {
-        val releaseNotesText = releaseNotes.asFile.get().readText()
-        if (!contributorSectionRegex.containsMatchIn(releaseNotesText)) {
-            throw IllegalStateException("Can't find the contributors section in the release notes $releaseNotes. You may need to update the regular expression.")
-        }
-        return contributorSectionRegex.find(releaseNotesText)!!.groupValues[2].lines()
+        val (_, contributorLines, _) = parseReleaseNotes()
+        return contributorLines
             .map { it.trim() }
             .filter { it.isNotEmpty() }
             .onEach { if (!contributorLineRegex.containsMatchIn(it)) throw IllegalStateException("Invalid contributor line: $it") }
             .map { GitHubUser(contributorLineRegex.find(it)!!.groupValues[2], contributorLineRegex.find(it)!!.groupValues[1]) }
             .toSet()
     }
 
+    protected
+    fun parseReleaseNotes(): Triple<List<String>, List<String>, List<String>> {
```

Comment:
Can you add some docs here what the three elements of the returned triple are? Or maybe even introduce a new data class instead of using a `Triple`?